### PR TITLE
AG8-136-cy-api-token-generate

### PR DESCRIPTION
**Pull request by nursena-tekoglu**
I am getting some failure, working on it

### CODE REVIEW

File: cypress/integration/step_definitions/nt_refresh_token.cy.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+})
+
+
+//Step definitions for Scenario: User refreshes their access token 
+
+
+Then ('a new access token is generated', () => {
+    
+    cy.refreshToken().then((response) => { 
+
```
Comment (KubraSA) [inline cypress/integration/step_definitions/nt_refresh_token.cy.js:25]: no need to call refreshToken function again. You re hitting end point again unnecessarily. Instead, take response in a global variable in previous function and use it here

Comment (nursena-tekoglu) [reply cypress/integration/step_definitions/nt_refresh_token.cy.js:25]: thank you for your feedback, I will update my code.

File: cypress/support/api.commands.js [hunk 2]
```diff
@@ -72,17 +72,11 @@ Cypress.Commands.add('upGradeDownGradeUsersubscription', (tier_type) => {
...
-  });
-   });    
-   
-   
\ No newline at end of file
+    url: Cypress.env('revayz_endpoint') + 'contentcreation', 
+    headers:  {
+      Authorization: 'Bearer ' + authToken
+    },
+    body,
```
Comment (KubraSA) [inline cypress/support/api.commands.js:79]: Does it accept body like that? It s not json or a json file from fixtures

File: cypress/e2e/features/backend/nt_refresh_token.feature [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+Scenario: User refreshes their access token 
+    Then a new access token is generated    
+    
+
+# Scenario: User tries to refresh with an invalid refresh token
+#     Given an invalid refresh token provided
+#     When the user requests a new access token using the refresh token feature
+#     Then an error response is returned with status code 404
+
```
Comment (KubraSA) [inline cypress/e2e/features/backend/nt_refresh_token.feature:14]: can you remove these commented scenarios unless you will execute them.

**Comment by nursena-tekoglu** [comment]
code completed and test is passed successfully @KubraSA

File: cypress/support/api.commands.js [hunk 4]
```diff
@@ -45,6 +45,33 @@ Cypress.Commands.add('userSubscription', () => {
 });
 
 
+//custom command to refresh token -nt
+
+Cypress.Commands.add ('refreshToken', () => { 
+
+  cy.fixture('refresh_token2').then((refresh_token_data) => {
+      var refresh_token2 = refresh_token_data.refresh_token;
+      cy.log(refresh_token2);
```
Comment (KubraSA) [inline cypress/support/api.commands.js:54]: can you remove this log?

File: cypress/support/api.commands.js [hunk 5]
```diff
@@ -45,6 +45,33 @@ Cypress.Commands.add('userSubscription', () => {
...
+      cy.request({
+          method: 'POST',
+          url: Cypress.env('revayz_endpoint') + 'token_generate',
+          body:  {
+            refresh_token: refresh_token2
+          }
+         })   
+         .then ((response) => { 
+          expect(response.body).to.have.property ('idToken');
+          expect(response.status).to.equal (200);
```
Comment (KubraSA) [inline cypress/support/api.commands.js:65]: we don't do assertions in supports. this should be done in test file

Comment (nursena-tekoglu) [reply cypress/support/api.commands.js]: I moved them

File: cypress/support/api.commands.js (hunk 4 above)
Comment (nursena-tekoglu) [reply cypress/support/api.commands.js]: I removed it

**Pull request merged**